concent_api/api-integration-multi-threads-test.py

```diff
@@ -0,0 +1,201 @@
+# !/usr/bin/env python3
+import os
+import sys
+import time
+from threading import Thread
+from typing import Optional
+
+import requests
+from golem_messages import message
+from golem_messages.factories.tasks import ReportComputedTaskFactory
+from golem_messages.message import Message
+from golem_messages.message.concents import ForceGetTaskResult
+from golem_messages.message.concents import ForceReportComputedTask
+from golem_messages.message.tasks import AckReportComputedTask
+from golem_messages.message.tasks import ReportComputedTask
+
+from api_testing_common import compare_lists_regardless_of_order
+from api_testing_common import count_fails
+from api_testing_common import PROVIDER_PRIVATE_KEY
+from api_testing_common import REQUESTOR_PRIVATE_KEY
+from api_testing_common import api_request
+from api_testing_common import create_signed_task_to_compute
+from api_testing_common import run_tests
+from api_testing_common import timestamp_to_isoformat
+from common.helpers import get_current_utc_timestamp
+from common.helpers import sign_message
+
+os.environ.setdefault("DJANGO_SETTINGS_MODULE", "concent_api.settings")
+
+REPORT_COMPUTED_TASK_SIZE = 10
+
+NUMBER_OF_TESTING_THREADS = 3  # changing this variable remember to correct number of expected responses
+MAXIMUM_WAITING_TIME_FOR_ALL_RESPONSES = 3  # seconds
+
+responses_global = []  # type: list
+
+
+def clear_responses():
+    global responses_global
+    responses_global.clear()
+
+
+def call_function_in_treads(
+    func,
+    number_of_threads: int,
+    cluster_url: str,
+    golem_message: Message,
+) -> None:
+    for i in range(number_of_threads):
+        thread = Thread(target=func, args=(cluster_url, golem_message,))
+        thread.start()
+
+
+def get_ack_report_computed_task(report_computed_task: ReportComputedTask) -> AckReportComputedTask:
```

Comment:
Why don't you use factories for this ? Each of those is used only once and does nothing extra except from passing arguments.

Reply:
I spoke with @kbeker about this and he said, it will be problem with keys. I don't remember exactly details, but this is the reason. I use factory at the beginning to create `report_computed_task`, and all next messages are like box for previous

Reply:
Ok, I thought it is safe to use those factories in general.

@kbeker Can you describe the problem with keys ?

Reply:
@rwrzesien 
Problem appears when we use `ReportComputedTask` factories. The point is that it gives to `provider_public_key` and `requestor_public_key` random keys and we don't have `private_key` for message serialization.
I checked and there is one work around for keys problems, when all keys are overriden in `ReportComputedTaskFactory` constructor.

Also, I think that if you use `AckReportComputedTask` factory here it might not be a problem because you have to override `report_computed_task` to have same parameters as in other messages.

Reply:
Ok, this is correct way.

But in this case I think it is easier and simpler to create 2 lines function:
> `def create_ack_report_computed_task(report_computed_task: ReportComputedTask) -> AckReportComputedTask:`
> `   return AckReportComputedTask(report_computed_task=report_computed_task)`

then use factory and change some fields. Factories are useful, I used them to create `ReportComputedTask`. But creating other messages is easier without factories. I would leave it as it is now..